common-app-lib/src/cmr/common_app/api/log.clj

```diff
@@ -0,0 +1,133 @@
+(ns cmr.common-app.api.log
+  "Defines the log routes for applications. There are two main functions 1) To get the current
+   CMR logging configuration and 2) Change the logging configuration. To get the logging
+   configuration from a service such as search locally one would use the following command:
+   curl -H \"Echo-Token: XXXX\" http://localhost:3003/log where the port goes to the correct service.
+   From the internet the command is:
+   curl -H \"Echo-Token: XXXX\" http://cmr.earthdata.nasa.gov/search/log
+   The above commands return the current CMR logging configuration to the caller. It will format the
+   message to either a basic html if one of the accepted headers is text/html, or as formatted raw text
+   if text/html is not included in the accept HTML header.
+   To change the configuration one needs to provide an EDN map of the configuration you want to replace.
+   For example if the configuraiton looks like the following:
+   {:level :warn,
```

Comment:
There's too much in this map. We should limit to only a subset of fields that we could change. For example we aren't going to be passing in functions. What fields should we actually allow modifying - from your example it looks like just `:level` and `:ns-pattern-map`?

Reply:
done
I allowed :level, :ns-pattern-map, :ns-blacklist, and :ns-whitelist.